### DISCUSSION TITLE
C-Code for Stein

### DISCUSSION THREAD

**Discussion by Dola47** [jeshraghian/snntorch#54] [Q&A]
Do we have any C code implementation for a single (Stein) Synaptic neuron?

Thanks.

**Comment by jeshraghian** [comment]
Not yet, would be great to implement the neuron models in C though.

**Reply by Dola47** [reply]
@jeshraghian I do not have any experience with C, but I just asked because I think at the end of the day when we have some implementation that needs to be implemented on the hardware it should be C-coded.  I will be glad if you can add such a feature on the Todo list, or support me in doing it by giving brief implementation ideas.

**Comment by Dola47** [comment]
@jeshraghian  I have one question related to the number of computations made inside of the ``Stein`` (Synaptic) neuron. 

Normally, for the conventional ANNs, we can use some of the available libraries to get to know how many GMacs we have in our network. 

However, I think for SNNs it is not only the GMacs done for example inside of a Spiking CNNs, but there should be also some computations related to the Spiking neuron itself (some computations done inside of the synaptic and the integration done on the spikes until we pass the threshold). 

I wonder have can we know the exact Multiplication-Additions (GMacs) we have in our Spiking Neural Network?

**Reply by Dola47** [reply]
Hi @jeshraghian  I hope that you have an answer to how can we precisely calculate the computations inside of the Spiking Neuron, and how can we calculate the correct overall computations within our SNN?

Thanks.
